### PR TITLE
Add queue_update_strategy TERMINATE and integration test

### PR DESCRIPTION
**Pull request by chenwany**
### Description of changes
Add TERMINATE value to QueueUpdateStrategy. With TERMINATE strategy, nodes will be put in POWER_DOWN_FORCR and replaced immediately, current running job in the replaced queue will be disrupted. Jobs in other queues will work as expected.

### Tests
* Unit tests
* Integration test

### References
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/update/test_update.py [hunk 1]
```diff
@@ -591,30 +613,40 @@ def _test_update_queue_strategy_with_running_job(
...
     queue1_nodes = scheduler_commands.get_compute_nodes("queue1")
     assert_compute_node_states(scheduler_commands, queue1_nodes, expected_states=["mixed", "allocated"])
-    # assert queue2 node state are in draining status
+    # check queue1 AMIs are not replaced
+    _check_queue_ami(cluster, ec2, pcluster_ami_id, "queue1")
+
     queue2_nodes = scheduler_commands.get_compute_nodes("queue2")
-    assert_compute_node_states(scheduler_commands, queue2_nodes, expected_states=["draining", "draining!"])
-    # retrieve queue2 instance again, check ami, they are not replaced
     ec2 = boto3.client("ec2", region)
```
Comment (lukeseawalker) [inline tests/integration-tests/tests/update/test_update.py]: duplicated, it was moved into line 630

File: tests/integration-tests/tests/update/test_update.py [hunk 2]
```diff
@@ -591,30 +613,40 @@ def _test_update_queue_strategy_with_running_job(
...
-    # requeue job in queue2 to launch new instances for nodes
-    remote_command_executor.run_remote_command(f"scontrol requeue {queue2_job_id}")
+    # assert queue2 node state are in expected status corresponding to the queue strategy
+    if queue_update_strategy == "DRAIN":
+        scheduler_commands.assert_job_state(queue2_job_id, "RUNNING")
+        _check_queue_ami(cluster, ec2, pcluster_ami_id, "queue2")
+        assert_compute_node_states(scheduler_commands, queue2_nodes, expected_states=["draining", "draining!"])
+        # requeue job in queue2 to launch new instances for nodes
+        remote_command_executor.run_remote_command(f"scontrol requeue {queue2_job_id}")
+    elif queue_update_strategy == "DOWN":
```
Comment (lukeseawalker) [inline tests/integration-tests/tests/update/test_update.py]: "DOWN" -> "TERMINATE"

File: tests/integration-tests/tests/update/test_update.py [hunk 3]
```diff
@@ -591,30 +613,40 @@ def _test_update_queue_strategy_with_running_job(
...
+        scheduler_commands.assert_job_state(queue2_job_id, "PENDING")
```
Comment (lukeseawalker) [inline tests/integration-tests/tests/update/test_update.py:646]: can you please confirm that jobs go directly into PENDING?

File: tests/integration-tests/tests/update/test_update.py (hunk 1 above)
Comment (chenwany) [reply tests/integration-tests/tests/update/test_update.py]: Good catch! Removed

File: tests/integration-tests/tests/update/test_update.py (hunk 2 above)
Comment (chenwany) [reply tests/integration-tests/tests/update/test_update.py]: Changed

File: tests/integration-tests/tests/update/test_update.py (hunk 3 above)
Comment (chenwany) [reply tests/integration-tests/tests/update/test_update.py:646]: Yes, after POWER_DOWN_FORCE on running job, the job turn from R to PD state

**Review by lukeseawalker — APPROVED** [review]

**Pull request merged**